src/pages/index.js

```diff
@@ -5,123 +5,260 @@ import PopupWithForm from "../components/PopupWithForm.js";
+    "Content-Type": "application/json",
+  },
+});
 
-const renderCard = (cardData) => {
-  const card = createCard(cardData);
-  section.addItem(card);
-};
+let section;
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
-}
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        cards,
+        renderer: (data) => {
+          const cardElement = createCard(data);
+          section.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-//Section.js
-//Section.js
+// UserInfo
+// UserInfo
+
+const userInfo = new UserInfo("#name", "#about", "#avatar");
 
-const section = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
-section.renderItems();
+api
+  .getProfileInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 // PopupWithForm
 // PopupWithForm
 
+const newCardPopup = new PopupWithForm({
+  popupSelector: "#add-card-modal",
+  handleFormSubmit: handleAddCardFormSubmit,
+});
+
 const editProfilePopup = new PopupWithForm({
   popupSelector: "#edit-modal",
   handleFormSubmit: handleProfileEditSubmit,
 });
-editProfilePopup.setEventListeners();
 
-const newCardPopup = new PopupWithForm({
-  popupSelector: "#add-card-modal",
-  handleFormSubmit: handleAddCardFormSubmit,
+const avatarPopup = new PopupWithForm({
+  popupSelector: "#profile-avatar-modal",
+  handleFormSubmit: handleAvatarFormSubmit,
 });
-newCardPopup.setEventListeners();
 
 // PopupWithImage
 // PopupWithImage
 
 const newImagePopup = new PopupWithImage("#image-preview-modal");
-newImagePopup.setEventListeners();
 
-// UserInfo
-// UserInfo
+// DeletePopup
+// DeletePopup
 
-const userInfo = new UserInfo(profileNameElement, jobElement);
+const cardDeletePopup = new DeleteConfirm("#delete-card", deleteCardForm);
 
-// Validation
-// Validation
+// Event Listeners
+// Event Listeners
 
-const editFormValidator = new FormValidator(settings, profileEditModal);
-const addFormValidator = new FormValidator(settings, addCardModal);
+profileFormElement.addEventListener("submit", (event) => {
+  event.preventDefault();
+});
+newImagePopup.setEventListeners();
+cardDeletePopup.setEventListeners();
+editProfilePopup.setEventListeners();
+newCardPopup.setEventListeners();
+avatarPopup.setEventListeners();
+
+// Form Validation
+// Form Validation
+
+const editFormValidator = new FormValidator(settings, profileFormElement);
 editFormValidator.enableValidation();
+
+const addFormValidator = new FormValidator(settings, addCardForm);
 addFormValidator.enableValidation();
 
-//Functions
-//Functions
+const avatarFormValidator = new FormValidator(settings, editAvatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```